src/main/java/org/cactoos/http/HtHead.java

```diff
@@ -57,50 +64,51 @@ public HtHead(final Input rsp) {
     }
 
     @Override
-    @SuppressWarnings("PMD.AvoidInstantiatingObjectsInLoops")
     public InputStream stream() throws IOException {
-        final InputStream stream = this.response.stream();
-        final byte[] buf = new byte[HtHead.LENGTH];
-        InputStream head = new DeadInputStream();
-        while (true) {
-            final int len = stream.read(buf);
-            if (len < 0) {
-                break;
-            }
-            final int tail = HtHead.findEnd(buf, len);
-            final byte[] temp = new byte[tail];
-            System.arraycopy(buf, 0, temp, 0, tail);
-            head = new SequenceInputStream(head, new InputStreamOf(temp));
-            if (tail != len) {
-                break;
-            }
-        }
-        return head;
+        final Scanner scanner = new Scanner(
+            this.response.stream(),
+            HtHead.CHARSET.name()
+        );
+        scanner.useDelimiter(HtHead.DELIMITER);
+        return new SequenceInputStream(
+            new HtHead.ScannerStreams(scanner)
+        );
     }
 
     /**
-     * Find header end.
-     * @param buf Buffer where to search
-     * @param len Size of the buffer
-     * @return End of the header
+     * Scanner to streams adapter.
      */
-    private static int findEnd(final byte[] buf, final int len) {
-        final byte[] end = {'\r', '\n', '\r', '\n'};
-        int tail = end.length - 1;
-        while (tail < len) {
-            boolean found = true;
-            for (int num = 0; num < end.length; ++num) {
-                if (end[num] != buf[tail - end.length + 1 + num]) {
-                    found = false;
-                    break;
-                }
-            }
-            if (found) {
-                tail = tail - end.length + 1;
-                break;
+    private static class ScannerStreams implements Enumeration<InputStream> {
+        /**
+         * Scanner instance.
+         */
+        private final Scanner scanner;
+
+        /**
+         * Ctor.
+         * @param scanner Scanner
+         */
+        ScannerStreams(final Scanner scanner) {
+            this.scanner = scanner;
+        }
+
+        @Override
+        public boolean hasMoreElements() {
+            return this.scanner.hasNext(".+");
+        }
+
+        @Override
+        public InputStream nextElement() {
+            final StringBuilder builder = new StringBuilder(
+                this.scanner.next()
+            );
+            if (this.hasMoreElements()) {
+                builder.append(HtHead.DELIMITER);
```

Comment:
I've checked against the previous commit 487af7 and 
Matchers.containsString("OK\r\nReferer") passes .
Which makes sense. Since the goal is to separate the HTTP header from the body so the separator between the headers needs to be kept.

Reply:
@ilyakharlamov ok yes, i better understand, thank you for confirming :)